Lib/http/client.py

```diff
@@ -918,21 +919,15 @@ def _tunnel(self):
         response = self.response_class(self.sock, method=self._method)
         (version, code, message) = response._read_status()
 
+        self._proxy_response_headers = parse_headers(response.fp)
+
+        if self.debuglevel > 0:
+            for hdr, val in self._proxy_response_headers.items():
+                print("header:", hdr + ":", val)
+
         if code != http.HTTPStatus.OK:
             self.close()
             raise OSError(f"Tunnel connection failed: {code} {message.strip()}")
-        while True:
-            line = response.fp.readline(_MAXLINE + 1)
-            if len(line) > _MAXLINE:
-                raise LineTooLong("header line")
-            if not line:
```

Comment:
I doubt this line. Was it needed here? Can a variable 'line' have some other falsy value other than an empty string?